src/main/java/net/minecraftforge/fmp/client/multipart/MultipartRegistryClient.java

```diff
@@ -0,0 +1,63 @@
+package net.minecraftforge.fmp.client.multipart;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.block.statemap.IStateMapper;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fmp.multipart.IMultipart;
+
+/**
+ * The clientside multipart registry. Allows you to bind a {@link MultipartSpecialRenderer} to a part.
+ *
+ * @see MultipartSpecialRenderer
+ */
+public class MultipartRegistryClient
+{
+    
+    private static Map<ResourceLocation, IStateMapper> specialMappers = new HashMap<ResourceLocation, IStateMapper>();
```

Comment:
The state mappers are unavoidable and they don't cause too big of an impact because they are only ran when the resources are reloaded. The MSRs, though... just use the same system as TESRs. I could make them use a string as an ID, but the difference won't be huge. If you have a better alternative, I'm open to suggestions :)